core/src/apps/cardano/helpers/utils.py

```diff
@@ -29,3 +33,17 @@ def variable_length_encode(number: int) -> bytes:
 def to_account_path(path: List[int]) -> List[int]:
     ACCOUNT_PATH_LENGTH = const(3)
     return path[:ACCOUNT_PATH_LENGTH]
+
+
+def fail_if_strict(fn, msg):
```

Comment:
I'm not sure which of these two approaches (`fail_if_strict` or `fail_if_true_and_strict`) is better, so I've included both. I prefer `fail_if_strict` since it doesn't clutter up the condition on which it should fail.